### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by stoljarjo**
- change function signature  for fmi2SampleDirectionalDerivative() in `FMI2_ext.jl`
- remove unnecessary functions in `FMI2_int.jl`
- change function fmi2DependenciesSupported() in `FMI2_md.jl`

### CODE REVIEW

File: src/FMI2_ext.jl [hunk 1]
```diff
@@ -418,15 +418,14 @@ end
...
                                           vUnknown_ref::Array{fmi2ValueReference},
                                           vKnown_ref::Array{fmi2ValueReference},
                                           dvUnknown::AbstractArray,
                                           steps::Union{Array{fmi2Real}, Nothing} = nothing)
     
-    if steps == nothing 
-        steps = fmi2GetReal(c, vKnown)
-        steps = abs.(steps) * eps(fmi2Real)
+    if steps === nothing 
+        steps = ones(fmi2Real, length(vKnown_ref)) .* eps(fmi2Real)
```
Comment (stoljarjo) [inline src/FMI2_ext.jl:428]: Please check

File: src/FMI2_md.jl [hunk 2]
```diff
@@ -712,12 +712,13 @@ end
 Returns if the FMU model description contains `dependency` information.
 """
 function fmi2DependenciesSupported(md::fmi2ModelDescription)
-    for mv in md.modelVariables
-        if mv.dependencies != nothing && length(mv.dependencies) > 0
-            return true
-        end
-    end 
+    der = md.modelStructure.derivatives
```
Comment (stoljarjo) [inline src/FMI2_md.jl:715]: Please check

File: src/FMI2_ext.jl (hunk 1 above)
Comment (ThummeTo) [reply src/FMI2_ext.jl:428]: should be `steps = abs.(fmi2GetReal(c, vKnown_ref)) * eps(fmi2Real)`

File: src/FMI2_md.jl (hunk 2 above)
Comment (ThummeTo) [reply src/FMI2_md.jl:715]: this should return if `depenendencies` are supported, not only `dependencies.derivatives`

**Review by ThummeTo — APPROVED** [review]

**Pull request merged**